CMakeLists.txt

```diff
@@ -13,6 +13,7 @@ endif ()
 # find packages
 find_package(fmt 9 COMPONENTS fmt-header-only QUIET)
 find_package(expected-lite 0.6 QUIET)
+find_package(Boost QUIET COMPONENTS)
```

Comment:
Please look if I did the boost dependency correctly, there are some things in the cmake logs that make it seem like its dragging in (maybe) unnecessary dependencies but idk how to fix that

Reply:
should be fine now. I've made the find_package REQUIRED. Otherwise, if we used cmake to get boost_multiprecision, we would be forced to export it what I don't want to do. 

Also, finding out which boost component includes what libraries is always a pain. Header seems. Side comment: header is  a bit odd. If you use the system package and cmake's own find_package definition, this is fine:
```
find_package(Boost REQUIRED COMPONENTS headers)
```

With conan it doesn't find headers here but linking against Boost::headers is fine.